### PR TITLE
Feat/logging overhaul

### PR DESCRIPTION
**Pull request by alikasim72**
Implementation of #1259. Replaced the old logging wrapper around the native python logging module.

### CODE REVIEW

**Comment by seanpearsonuk** [comment]
@alikasim72 I haven't read every single line, but it looks good. I also haven't looked into the CI. Are these random issues again, or something triggered by the changes?

**Comment by alikasim72** [comment]
@seanpearsonuk these were some random minor errors caused by me using absolute file location, fixed it by using the paths from the os library.

File: src/ansys/fluent/core/logging_config.yaml
Comment (mkundu1) [inline src/ansys/fluent/core/logging_config.yaml:1]: Could you please check if this is packaged? You may need to add this in MANIFEST.in

Comment (alikasim72) [reply src/ansys/fluent/core/logging_config.yaml:1]: Good Point!

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/__init__.py [hunk 1]
```diff
@@ -44,44 +44,16 @@ def version_info() -> str:
...
-
-def disable_logging_to_file() -> None:
-    """Disable logging to file."""
-    LOG.disable_logging_to_file()
+# Load the logging configuration from a YAML file
+with open(yaml_path, "rt") as f:
+    config = yaml.safe_load(f)
 
+# Configure the logging system
+logging.config.dictConfig(config)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/__init__.py:56]: @alikasim72 I looked into the question of what happens if this method is called more than once (we are just one library):
https://stackoverflow.com/a/47496535

So, different configs cannot play together?

**Review by mkundu1 — APPROVED** [review]

**Pull request merged**